src/gdbserver/gdb-server.c

```diff
@@ -38,7 +38,7 @@
 static stlink_t *connected_stlink = NULL;
 static bool semihosting = false;
 static bool serial_specified = false;
-static char serialnumber[28] = {0};
+static uint8_t serialnumber[STLINK_SERIAL_SIZE] = {0}; // XXX? Why was size 28?
```

Comment:
Be carefull with this. The discovery boards (stlinkv2) have a binary serial number (12bytes). The nucleo boards (stlinkv2-1) have a hex encoded serial number (probably 24 or 25 bytes not 100% sure right now).